include/fastrtps/Domain.h

```diff
@@ -184,6 +185,7 @@ class Domain
 
 private:
 
+    static std::recursive_mutex m_mutex;
```

Comment:
Why recursive mutex at this level? With a normal mutex is enough.

Reply:
You are right. Changed.

Reply:
Not sure if that is correct. `Domain::stopAll()` calls `Domain::removeParticipant()` in a loop. And that second method also acquires the lock. 
Because `Domain::stopAll()`  loops over `m_participants.size()` and that is a shared variable, I think a recursive mutex is required...

Reply:
@guillaumeautran we have changed this method in a second push to just remove all the elements of the list of participants without calling `Domain::removeParticipant()`, because these searches are not needed.

Reply:
I see it now. Missed that change before though. 
Yes, in that case, a recursive mutex is not needed. Thanks!